### ISSUE TITLE
Allow null value of aria-activedescendent when element does not have DOM focus

### ISSUE THREAD

**Issue by jnurthen** [w3c/aria#501]
It is common practice when creating widgets using `aria-activedescendent` to set `aria-activedescendent=""` until the component gets DOM focus.
This is currently not allowed by the spec as `aria-activedescendent` needs to be an IDREF

`aria-activedescendent` is only relevant when the component has DOM focus as such I propose that we allow an undefined value when the component does not have focus.

I prefer this to the alternative of developers adding the attribute and its value the first time it gets focus, as it gives an easy way to browse the DOM in inspection tools and easily know which type of focus model each composite widget is using - even before they have been focused.

**Comment by mcking65** [comment]
Why not even if it has focus?

Consider a combo where activedescendant is set on the text field and DOM focus is on the text field. When typing in the text field, activedescendant is null but when down arrow is pressed then it gets set to the idref of an element in the popup. Then, there can be a variety of conditions where we want AT focus to be back on the text field and activedescendant can be null again. The entire time, DOM focus is on the text field.

It seems like allowing null values in this case would be useful.

**Comment by AmeliaBR** [comment]
This seems reasonable, but a question:

How would you describe the semantic meaning of having the attribute with an empty value?  A kind of flag that this element _can_ have active sub-components, even if none of them are currently active?  Or should it just be treated as equivalent to the attribute not being there?

**Comment by mcking65** [comment]
[ARIA Section  7.6 State and Property Attribute Processing]() 
states:

> Sometimes states and properties are present in the DOM but have a zero-length string ("") as their value. This is equivalent to their absence. User agents should treat state and property attributes with a value of "" the same as they treat an absent attribute. For supported states and properties, this corresponds to the default value, but if it is a required attribute, it signals an author error, and the implicit value for the role is used.

Since aria-activedescendant is never a required property, shouldn't the empty string be treated as if the attribute is not present?

I don't see anything in the spec that disallows an undefined value for aria-activedescendant or any other non-required property that has an IDREF as its value.
